lib/Backend/arm64/LowerMD.cpp

```diff
@@ -8372,6 +8317,35 @@ void LowererMD::GenerateFloatTest(IR::RegOpnd * opndSrc, IR::Instr * insertInstr
     insertInstr->InsertBefore(instr);
 }
 
+IR::RegOpnd* LowererMD::CheckFloatAndUntag(IR::RegOpnd * opndSrc, IR::Instr * insertInstr, IR::LabelInstr* labelHelper)
+{
+    IR::Opnd* floatTag = IR::IntConstOpnd::New(Js::FloatTag_Value, TyMachReg, this->m_func, /* dontEncode = */ true);
+
+    // MOV floatTagReg, FloatTag_Value
+    if (!opndSrc->GetValueType().IsFloat())
+    {
+        // TST s1, floatTagReg
+        IR::Instr* instr = IR::Instr::New(Js::OpCode::TST, this->m_func);
+        instr->SetSrc1(opndSrc);
+        instr->SetSrc2(floatTag);
+        insertInstr->InsertBefore(instr);
+        LegalizeMD::LegalizeInstr(instr, false);
+
+        // BZ $helper
+        instr = IR::BranchInstr::New(Js::OpCode::BEQ /* BZ */, labelHelper, this->m_func);
+        insertInstr->InsertBefore(instr);
+    }
+
+    IR::RegOpnd* untaggedFloat = IR::RegOpnd::New(TyMachReg, this->m_func);
+    IR::Instr* instr = IR::Instr::New(Js::OpCode::EOR, untaggedFloat, floatTag, opndSrc, this->m_func);
```

Comment:
Think you need to swap floatTag and opndSrc here, unless the system is smart enough to move the immediate for a commutative operation.